#!/usr/bin/env python
PACKAGE = "morus_control"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("vz_kp", double_t, 0, "VZ ctl PID P gain", 40, 0, 100)
gen.add("vz_ki", double_t, 0, "VZ ctl PID I gain", 0.1, 0, 100)
gen.add("vz_kd", double_t, 0, "VZ ctl PID D gain", 0.0, 0, 100)

gen.add("z_kp", double_t, 0, "Z ctl PID P gain", 4, 0, 100)
gen.add("z_ki", double_t, 0, "Z ctl PID I gain", 0.0, 0, 100)
gen.add("z_kd", double_t, 0, "Z ctl PID D gain", 1.0, 0, 100)

gen.add("yaw_kp", double_t, 0, "Yaw ctl PID P gain", 1, 0, 100)
gen.add("yaw_ki", double_t, 0, "Yaw tl PID I gain", 0.0, 0, 100)
gen.add("yaw_kd", double_t, 0, "Yaw ctl PID D gain", 0.0, 0, 100)

gen.add("yaw_rate_kp", double_t, 0, "Yaw rate ctl PID P gain", 400, 0, 800)
gen.add("yaw_rate_ki", double_t, 0, "Yaw rate ctl PID I gain", 0.0, 0, 100)
gen.add("yaw_rate_kd", double_t, 0, "Yaw rate ctl PID D gain", 0.0, 0, 100)

gen.add("z_lambda", double_t, 0, "Z ctl compensator lambda", 0.1, 0, 10)
gen.add("vz_lambda", double_t, 0, "VZ ctl compensator lambda", 2, 0, 50)

gen.add("yaw_lambda", double_t, 0, "Yaw ctl compensator lambda", 0.1, 0, 10)
gen.add("yaw_rate_lambda", double_t, 0, "Yaw rate ctl compensator lambda", 0.1, 0, 10)

gen.add("z_comp_gain", double_t, 0, "Z compensator gain", 0.05, 0, 50)
gen.add("vz_comp_gain", double_t, 0, "VZ compensator gain", 100, 0, 300)

gen.add("yaw_comp_gain", double_t, 0, "Yaw compensator gain", 0.05, 0, 10)
gen.add("yaw_rate_comp_gain", double_t, 0, "Yaw rate compensator gain", 0.1, 0, 100)

gen.add("switch_eps", double_t, 0, "Switching constant", 0.01, 0, 1)
gen.add("z_beta", double_t, 0, "Z switch gain", 0.01, 0, 1)
gen.add("vz_beta", double_t, 0, "VZ switch gain", 30, 0, 200)

gen.add("yaw_beta", double_t, 0, "Yaw switch gain", 0.01, 0, 1)
gen.add("yaw_rate_beta", double_t, 0, "Yaw rate switch gain", 1, 0, 100)

gen.add("z_ff", double_t, 0, "Z feed-forward gain", 1, 0, 10)
gen.add("vz_ff", double_t, 0, "VZ feed-forward gain", 8, 0, 50)

gen.add("yaw_ff", double_t, 0, "Yaw feed-forward gain", 0.01, 0, 1)
gen.add("yaw_rate_ff", double_t, 0, "Yaw rate feed-forward gain", 0.01, 0, 1)

exit(gen.generate(PACKAGE, "morus_control", "SmcUavHeightCtlParams"))
